### ISSUE TITLE
Verify learning rate is being used correctly

### ISSUE THREAD

**Issue by motiwari** [ThrunGroup/FastForest#124]
Do you have a reference for the learning rate being used as it is in [this line](https://github.com/ThrunGroup/FastForest/blob/main/utils/utils.py#L286)?

Most of the references I'm finding suggest an alternative use of the learning rate parameter:
a) Fit the new model to the full (unmodified) residual of the current model's predictions vs. the true labels
b) Add the models as `old_model + learning_rate*new_model` (see [here](https://scikit-learn.org/stable/modules/ensemble.html#shrinkage-via-learning-rate)

in which case we will have to update our use of the learning rate
